### PR TITLE
Image Model Updated (Issue Fixed #15 )

### LINKED ISSUE #15: Feature: Add size attribute to "Image" models and store image size inside it.
* Add migration to create size attribute on Image model
* Add attribute to image model and ensure it wills store the size in kb on the "image" currently referenced by it.

This can be used to estimate size on disk of each campaign.

### PR DESCRIPTION
**Pull request by devjj248**
"Size"attribute successfully added. Also size storing the Image table via Image Model
sizeAttributeadded

### CODE REVIEW

File: app/Models/Campaign/Image.php [hunk 1]
```diff
@@ -40,9 +41,14 @@ public static function upload($campaign, $img, $name = null)
         return DB::transaction(function() use ($campaign, $img, $name) {
             $path = static::uploadImage($campaign, $img);
             $title = ($name) ? $name : $img->getClientOriginalName();
-
+            $file_name = $img->getClientOriginalName();
```
Comment (thybag) [inline app/Models/Campaign/Image.php]: Hi @junaidJaved8461 Cheers for contributing!

In this case i believe the  `$img` should be an instance of `Illuminate\Http\UploadedFile`  so you shouldn't need to grab/move the file at all here. In theory type & size should be grabbable via the `->getMimeType()` and `getSize()`.

In terms of storage, it'd be best to store without the `kb` if possible also, that way we can easily aggregate the size of all the uploaded files on a campaign for example, by calling sum on the sizes. (The kb part can then be added in the UI, or accessor if we always want it).

**Review by thybag — CHANGES_REQUESTED** [review]
Hi @JunaidJaved8461 Cheers for contributing!

Had an initial read through, and think there may be an easier way to grab the size info. It'd also be good to include the paid migration within this PR - so it can be merged as a functional unit etc.

Hope my feedback all makes sense.
Bag

File: app/Models/Campaign/Image.php [hunk 2]
```diff
@@ -40,9 +41,14 @@ public static function upload($campaign, $img, $name = null)
...
+            $type = $img->getClientOriginalExtension();
+            $file_path =  $img -> move(public_path().'/upload/images', $file_name);
```
Comment (thybag) [inline app/Models/Campaign/Image.php]: the spaces here will likely throw an error. The image should already be stored via `uploadImage` so not sure this is needed at all. Everything is stored within the user_id/campaign_id/ directory to make cleaning up easier. (You may need to run the link filesystem artisan command if they don't show up for you)

**Comment by devjj248** [comment]
without kb
Hi @thybag  Please review the latest commits & gave me feedback & let me know if any changes require!! Thanks

**Review by devjj248 — COMMENTED** [review]
Requested Changes Done!! Let me Know when you review my commits

**Comment by devjj248** [comment]
@thybag Have you checked my PR after latest commits?

File: app/Models/Campaign/Image.php [hunk 3]
```diff
@@ -40,9 +42,12 @@ public static function upload($campaign, $img, $name = null)
         return DB::transaction(function() use ($campaign, $img, $name) {
             $path = static::uploadImage($campaign, $img);
             $title = ($name) ? $name : $img->getClientOriginalName();
-
+            $file_name = $img->getClientOriginalName();
```
Comment (thybag) [inline app/Models/Campaign/Image.php]:
```suggestion
```
This appears unused

**Comment by devjj248** [comment]
@thybag All Okay then merge it otherwise let me know if any more change required!! Thanks

**Pull request merged**